src/hpa.c

```diff
@@ -144,6 +144,9 @@ hpa_central_extract(tsdn_t *tsdn, hpa_central_t *central, size_t size,
 			*oom = true;
 			malloc_mutex_unlock(tsdn, &central->grow_mtx);
 			return NULL;
+		} else if (start_as_huge) {
+			central->hooks.hugify(new_eden, HPA_EDEN_SIZE,
```

Comment:
Based on our discussions offline, I thought we want to:
1. change the call to `pages_map` above to a hook;
2. when `start_as_huge == true`, call another hook that does both map and hugification.

Is that still the case or you want to do it later? I am fine if we want to keep the refactoring separate as well.